relay-profiling/src/transaction_metadata.rs

```diff
@@ -0,0 +1,56 @@
+use serde::{Deserialize, Serialize};
+
+use relay_general::protocol::EventId;
+
+use crate::utils::deserialize_number_from_string;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct TransactionMetadata {
+    pub id: EventId,
+    pub name: String,
+    pub trace_id: EventId,
+
+    #[serde(deserialize_with = "deserialize_number_from_string")]
+    pub relative_start_ns: u64,
+
+    #[serde(deserialize_with = "deserialize_number_from_string")]
+    pub relative_end_ns: u64,
+}
+
+impl TransactionMetadata {
+    pub fn valid(self) -> bool {
```

Comment:
```suggestion
    pub fn valid(&self) -> bool {
```

I don't think `valid` needs to take ownership of the instance here. 

It's just reading and checking if it's valid.

This way you won't have to clone every time just to check if the `TransactionMetadata` is valid as currently done [here](https://github.com/getsentry/relay/pull/1444/files#diff-3be70af25bca3e9b8f83cb0bdd32c85b2755fabbbe802f77c3c1c174df13fc57R206)